ili/validation/metrics.py

```diff
@@ -508,6 +509,64 @@ def _plot_TARP(
         fig.savefig(filepath)
         return fig
 
+    def _calc_true_logprob(
+        self, samples: np.array, trues: np.array,
+        signature: str, bw_method: str = "scott"
+    ) -> np.array:
+        """Calculate the probability of the true parameters under the
+        learned posterior.
+
+        Notes:
+            This is implemented by using a Gaussian KDE as a variational
+            distribution for the posterior, constructed from the samples. If
+            there are not enough samples, not enough test points, or there are
+            sharp priors, the KDE may be inaccurate.
+
+        Args:
+            samples (np.array): posterior samples of shape (nsamples, ndata, npars)
+            trues (np.array): true parameters of shape (ndata, npars)
```

Comment:
Please add docstrings for the optional arguments

Reply:
Added.